### PR TITLE
fix: trigger iframe reload before show

### PR DESCRIPTION
**Pull request by dhavkuma**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://main--dx-recognitions--hlxscreens.hlx.live/content/screens/org-amitabh/dashboards/ban2-08-130
- After: https://live--dx-recognitions--hlxscreens.hlx.live/content/screens/org-amitabh/dashboards/ban2-08-130

### CODE REVIEW

File: blocks/carousel/carousel.js [hunk 1]
```diff
@@ -300,6 +309,7 @@ export default async function decorate(block) {
     if (!isActive(currentIndex)) {
       nextSlide();
     } else {
+      reloadSlide(currentIndex);
```
Comment (Danialkafeel) [inline blocks/carousel/carousel.js:312]: lets add a check here that if current slide is a dashboard iframe, then only `reloadSlide` is called

File: blocks/carousel/carousel.js [hunk 2]
```diff
@@ -283,6 +283,15 @@ export default async function decorate(block) {
     return false;
   }
 
+  function reloadSlide(itemIndex) {
+    if (itemIndex < 0 || itemIndex >= totalItems) {
+      return;
+    }
+    const currentItem = carouselItems[itemIndex];
+    const currentIframe = currentItem.querySelector('iframe');
+    currentIframe.src = currentIframe.src; // reassigning src will reload iframe
```
Comment (Danialkafeel) [inline blocks/carousel/carousel.js:292]: Please add a `null` check here

**Comment by Danialkafeel** [comment]
@dhavkuma Lets try to update the logic to reload the next slide. This way any delays are avoided and every slide comes updated.
Similar to what we do for videos in screens-core

File: blocks/carousel/carousel.js (hunk 1 above)
Comment (dhavkuma) [reply blocks/carousel/carousel.js:312]: recongnitions are not iframe. So no need to add this check.

Comment (dhavkuma) [reply blocks/carousel/carousel.js:312]: I intend to reload any iframe agnostic of use case.

File: blocks/carousel/carousel.js [hunk 3]
```diff
@@ -293,13 +320,19 @@ export default async function decorate(block) {
...
+  function preloadNextSlide(itemIndex) {
+    const nextItemIndex = getNextItemIndex(itemIndex);
+    reloadSlide(nextItemIndex);
+  }
+
   function nextSlide() {
     // Stop the previous carousels
     TIMEOUTS.clearAllTimeouts();
-    currentIndex = (currentIndex + 1) % totalItems;
+    currentIndex = getNextItemIndex(currentIndex, totalItems);
```
Comment (Danialkafeel) [inline blocks/carousel/carousel.js:331]: passing two arguments here but function is receiving one only

File: blocks/carousel/carousel.js [hunk 4]
```diff
@@ -283,6 +287,29 @@ export default async function decorate(block) {
...
+  function reloadIframe(currentIframe) {
+    if (currentIframe && currentIframe.src) {
+      currentIframe.src = currentIframe.src; // reassigning src will reload iframe
+    }
+  }
+
+  function reloadSlide(itemIndex) {
+    function getIframeInSlide(index) {
+      if (index < 0 || index >= totalItems) {
+        return;
```
Comment (Danialkafeel) [inline blocks/carousel/carousel.js:303]: should `null` be returned from here?

File: blocks/carousel/carousel.js [hunk 5]
```diff
@@ -283,6 +287,29 @@ export default async function decorate(block) {
     return false;
   }
 
+  function getNextItemIndex(itemIndex) {
+    return (itemIndex + 1) % totalItems;
+  }
+
+  function reloadIframe(currentIframe) {
```
Comment (Danialkafeel) [inline blocks/carousel/carousel.js:294]: lets rename `currentIframe` to `iframe`, names seems incorrect with updated logic

**Pull request merged**